### DISCUSSION TITLE
Can internal updates of APK applications be implemented?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#3277] [Q&A]
Is there any mechanism to update APK internally?

Option 1: Consider using load to load SWFs in the main program, and only update these SWFs during updates. However, this way, classes like File cannot be used in the loaded SWFs, making compilation inconvenient. If there are any changes that require the use of AIR method classes, APK still needs to be reinstalled.
Option 2: I have been trying to place the APK on the server and update it by downloading it, but this method requires redirecting to the browser, which is not very convenient to update.

Is there a way to dynamically download and update the main SWF file? Perhaps I am not aware of such a method, so please come and seek advice. Thank you!

**Comment by marchbold** [comment]
You can use the Play Store or Huawei App-Gallery to facilitate in app updates:

https://docs.airnativeextensions.com/docs/inappbilling/in-app-updates

This process allows your app to check for updates and display a store managed update dialog within your application and then download and update as required.

**Reply by Jason0247** [reply]
App discussions are indeed a good solution, but they need to be validated every time they update, which takes a long time.

**Reply by marchbold** [reply]
Are you talking about bypassing the app store review process? I believe this sort of action will get your account revoked. Might be worth checking the terms before going too far
